Add attributes to waypoint and route documents

asaunier opened the pull request:
![screenshot from 2016-02-17 14 58 13](https://cloud.githubusercontent.com/assets/7814311/13111613/1047f172-d587-11e5-9651-8de6303322bb.png)
![screenshot from 2016-02-17 14 58 38](https://cloud.githubusercontent.com/assets/7814311/13111616/122256d6-d587-11e5-9da6-fb8215c6eb8b.png)
![screenshot from 2016-02-17 14 59 04](https://cloud.githubusercontent.com/assets/7814311/13111618/193079a8-d587-11e5-8d51-bd2160ed5f35.png)
![screenshot from 2016-02-17 14 59 10](https://cloud.githubusercontent.com/assets/
![screenshot from 2016-02-17 14 59 20](https://cloud.githubusercontent.com/assets/7814311/13111627/20217f14-d587-11e5-8d59-ff4b2dcdf166.png)
7814311/13111624/1d7d4590-d587-11e5-900a-71f7a211ed0b.png)
![screenshot from 2016-02-17 14 59 10](https://cloud.githubusercontent.com/assets/7814311/13111647/381cb32c-d587-11e5-9d8c-196c8d3ff365.png)

![screenshot from 2016-02-17 14 59 20](https://cloud.githubusercontent.com/assets/7814311/13111630/2426ce52-d587-11e5-910d-d8ff0ec0d993.png)

Route and waypoints details-view updated. Clearly there is much more work to do and things to make better but I would like to merge this PR as is and make updates in future PRs. It would be great to merge this and then side menu PR.

- view details directive for clicks on tabs and for modals
- responsive-floatingbuttons.js is no more required (Deleted)

Todo : 
[ ] different attributes for different route types
[ ] show only attributes that have data, but still show some of the most important with 'no data available'
[ ] create generic helpers in order to factorize the views and attributes

asaunier commented:
@ginold I have made changes to your initial commits (that added helpers to display the "figures" attributes):
* since we now have many helpers functions I have dispatched them in several files (and add some indentation inside the functions in order to make the files more readable)
* I have factorized your 3 helpers into a single one (using parameters).

I have noticed that the content of the documents are added twice in the HTML (1 for big screens, 1 for small screens). The layout is indeed very nice and handy (using tabs for small screens) but I don't think having the content duplicated is acceptable => it kind of double the size of the HTML code (think that some document really have a lot of content, see for instance http://www.camptocamp.org/summits/107049/fr/presles-escalade). Not to mention it duplicates the templates actions.

I really think we must have the content only once, even though it means that the layout is not that cool for small screens. By the way, we're supposed to follow the layout described in the wireframes, not to recreate everything from scratch :P 

Could you fix this? Thanks!

asaunier commented:
Another point: in this PR, "figures" attributes are displayed independently. That's indeed nice and simple.
But that takes a lot of room. Some attributes will be grouped (using icons?). For instance the min/max elevations of a route, its height diff up/down, the grades, etc.

asaunier commented:
My comments as a user:
* OH-MY-GOD THIS-IS-AWESOME!!!! => the layout and the responsiveness is excellent and match really well the wireframes
* The "See in other language" at the end of the page is a bit useless (and hidden) since already available in the side tools.
* In my instance, when the modal box does not open as expected when clicking on "See in other language" in tool "More"!?
* I get a mako error when loading http://c2corgv6-demo.gis.internal:6555/routes/50709/fr because
```
File "/home/alex/v6_ui/c2corg_ui/templates/route/view.html", line 130, in render_body
    % if route['equipment_rating']:
KeyError: 'equipment_rating'
```
=> this field is only available for climbing routes (not skiing etc.)
* I also have a mako error when visiting an archived version of the page. For instance http://c2corgv6-demo.gis.internal:6555/waypoints/37355/fr/1021748
```
File "/home/alex/v6_ui/c2corg_ui/templates/waypoint/view.html", line 73, in render_body
    <li>${show_archive_data('routes', route, locale, version)}</li>
  File "/home/alex/v6_ui/c2corg_ui/templates/helpers.html", line 60, in render_show_archive_data
    <li><a href="${request.route_url(module + '_view', id=document['document_id'], lang=locale['lang'])}" translate>See the latest version</a></li>
TypeError: 'Undefined' object is not subscriptable
```

asaunier commented:
Concerning the "archived version" page, there are actually 2 errors:
* there's a typo at https://github.com/c2corg/v6_ui/blob/waypoint_route_attributes/c2corg_ui/templates/waypoint/view.html#L73 => should be "waypoints"/"waypoint" instead of "routes"/"route"
* associated docs should not be displayed when viewing an "archived version" (not sure the API returns associated docs when asking for an archived version) => this is not an issue from your  PR, probably comes from the PR about the associations :P

tsauerwein commented:
>  (not sure the API returns associated docs when asking for an archived version)

No, it should not.

asaunier commented:
Thanks for clarifying @tsauerwein The error actually occurs with branch "master". I have created an issue: https://github.com/c2corg/v6_ui/issues/172
No need to handle this issue in this PR.

asaunier commented:
Since "areas" are now listed in the "figures" block at the beginning. You may remove them for the associations lists at the end of the document.
What about doing the same for associated maps?

asaunier commented:
Thanks for adding the "view in another lang" tool. The modal actually appears but nothing (no redirection) happens when clicking on a language?

asaunier commented on c2corg_ui/static/js/detailsview.js:
```diff
@@ -0,0 +1,69 @@
+
+/**
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
```
Not sure the ``@ngInject`` is needed here.

asaunier commented on c2corg_ui/static/js/detailsview.js:
```diff
@@ -0,0 +1,69 @@
+    link: function(el, scope, attrs) {
+      // show tabs if they have been hidden on smartphone
+      $(window).resize(function() {
+        if (window.innerWidth > 620) {
```
What about listing those width params in some JS file where they would be centralized?
I'm a bit concerned that if one day we want to change those values, we would have to search/replace in many files and possibly struggle because some setting would have been forgotten...

asaunier commented on c2corg_ui/static/js/detailsview.js:
```diff
@@ -0,0 +1,69 @@
+ */
+app.ViewDetailsController.prototype.openTab = function(tab) {
+  // only for smartphones
+  if (window.innerWidth < 620) {
```
Same comment here for the width value

asaunier commented on c2corg_ui/templates/document/diff.html:
```diff
@@ -1,9 +1,12 @@
-<%namespace file="../helpers.html" import="show_version_comment"/>
+
+<%namespace file="../helpers/common.html" import="show_title"/>
+
```
Do we need those extra blank lines?

asaunier commented on c2corg_ui/templates/helpers.html:
```diff
@@ -61,37 +61,6 @@
   </ul>
```
This file ``c2corg_ui/templates/helpers.html`` should be removed since splitted in several ``c2corg_ui/templates/helpers/*.html`` files.

asaunier commented on c2corg_ui/templates/helpers/view.html:
```diff
@@ -0,0 +1,117 @@
+  ${'{{mainCtrl.translate(\'' + get_attr(version, 'comment', md=False) + '\')}}' if version['comment'] else '' | n}
+</%def>
+
+<%def name="create_attribute_element_value(type, attr)">\
```
Those 3 helpers seem no longer used (replaced by ``show_short_attr`` defined a little above in the file).

asaunier commented on c2corg_ui/templates/route/view.html:
```diff
@@ -1,8 +1,10 @@
 from c2corg_ui.templates.utils import get_lang_lists
 %>\
+<%namespace file="../helpers/view.html" import="show_attr, show_short_attr, show_missing_langs_links, show_other_langs_links"/>
+
```
It would make more sense to move this ``<%namespace`` after the ``<%inherit`` near the other ``<%namespace``.

asaunier commented on c2corg_ui/templates/route/view.html:
```diff
@@ -17,65 +19,162 @@
+        <p translate>Access</p>
+      </div>
+      <span class="detail-text accordion" ng-init="lift_access=${route['lift_access']}">
+        <span translate >Approche en remontée mécanique</span>: 
```
Extra space after "translate" :p 
Please use an english string for the attribute title.

asaunier commented on c2corg_ui/templates/route/view.html:
```diff
@@ -17,65 +19,162 @@
     <span class="lead"> 
       <h3 translate class="text-center heading">description</h3>
-      ${show_attr(locale, 'description')}
+        ${show_attr(locale, 'description')}
```
Indent not really needed here.

asaunier commented on c2corg_ui/templates/route/view.html:
```diff
@@ -17,65 +19,162 @@
+   ## IMAGES
+  <div class="view-details-photos col-xs-12 tab">
+    <span class="lead"> 
+      <h3 translate class="text-center ">images</h3>
```
extra space

asaunier commented on c2corg_ui/templates/route/view.html:
```diff
@@ -99,5 +198,15 @@ <h3 class="text-center heading">Associations</h3>
+  </div>
 </section>
-<%include file="../document/floating-button-responsive.html"/>
+
```
Extra blank lines!

asaunier commented:
I have rebased the PR again master (there were a bunch of unrelated commits :P) and commited a few changes (thus addressing most of the comments I made earlier):
* the splitting of ``helpers.html`` had suffered from some conflicts fixes. Fixed at f3e994b
* associated docs are no longer listed when viewing an archived version. Fixes https://github.com/c2corg/v6_ui/issues/172
* I have moved the detail view side/bottom tools at the end of the HTML (remain at the same place in the rendered page) because it's better to have the main content or main navigation links at the beginning of the document (for loading questions or when using page-reading tools for instance). See d16282a

@tsauerwein @gberaudo Do you have any objection against merging this PR?

From my perspective it's OK for merging even though I have still a few remarks (can be adressed in a separated PR):
* I think we should avoid having "hard-coded" screen widths in the directives. See issue https://github.com/c2corg/v6_ui/issues/173
* for archived versions:
 * It would be better to display the "You are viewing an archived version of this document." warning at the beginning of the page (just before the description). I know it's shown in the "more" tool (as specified in the wireframes?) but it's way to discrete, people won't notice it and will try to edit the document thinking it's out of date. I think you may remove the warning from the "more" tool. By the way the warning is also displayed (mistake I guess) at the end of archived waypoint pages.
 * we don't need the "translate in another language" action in the "more" tool when viewing an archived version

tsauerwein commented on c2corg_ui/static/js/detailsview.js:
```diff
@@ -0,0 +1,68 @@
+      // show tabs if they have been hidden on smartphone
+      $(window).resize(function() {
+        if (window.innerWidth > 620) {
+          $('.tab, .accordion').show();
```
If no new elements with a class `.tab` or `.accordion` are added, you could cache the selection:

    var elements = $('.tab, .accordion');
    $(window).resize(function() {
      if (window.innerWidth > 620) {
        elements.show();

tsauerwein commented:
> It would be better to display the "You are viewing an archived version of this document." warning at the beginning of the page (just before the description). I know it's shown in the "more" tool (as specified in the wireframes?) but it's way to discrete, people won't notice it and will try to edit the document thinking it's out of date

Makes sense. 

I only skimmed through the PR, the screenshots are looking promising!

asaunier commented:
Thanks. looks good, please merge when Travis passes.
